lib/membrane_rtmp_plugin/rtmp/source/message_parser.ex

```diff
@@ -149,14 +149,19 @@ defmodule Membrane.RTMP.MessageParser do
             # needs to be stripped and is not counted into the body_size
             headers_to_strip = div(body_size - 1, chunk_size)
 
-            body_size + headers_to_strip
+            timestamps_to_strip = if header.extended_timestamp?, do: headers_to_strip * 4, else: 0
+            body_size + headers_to_strip + timestamps_to_strip
           else
             body_size
           end
 
         case rest do
           <<body::binary-size(chunked_body_size), rest::binary>> ->
-            combined_body = combine_body_chunks(body, chunk_size)
+            combined_body = combine_body_chunks(body, chunk_size, header)
+
+            if byte_size(combined_body) != body_size do
```

Comment:
This guard was very helpful to make sure that we passed the correct body to the deserialize_message function. Not sure if we should keep it or not. It currently makes credo fail. Let me know if I should remove it or extract the check into a private assert function.